python/cudf/cudf/_lib/pylibcudf/__init__.pxd

```diff
@@ -1,9 +1,13 @@
 # Copyright (c) 2023, NVIDIA CORPORATION.
 
 # TODO: Verify consistent usage of relative/absolute imports in pylibcudf.
-from . cimport copying
+# TODO: Cannot import interop because it introduces a build-time pyarrow header
+# dependency for everything that cimports pylibcudf. See if there's a way to
+# avoid that before polluting the whole package.
+from . cimport copying  # , interop
```

Comment:
I think that's fair, and I agree that's how you'd solve it in C/C++. It's a question of how you'd want a Cython package to look; in Python it feels more natural to import a package and be able to access subpackages/submodules without needing to do a separate import, wouldn't you say?

For now this particular question is no longer imminently pressing. The changes to move the interop code directly into Column/Scalar means that all of cudf's Cython now requires these headers anyway, so for consistency I'll do the same in pylibcudf since it's no longer so onerous relatively speaking. I definitely want to come back to this discussion when pylibcudf is closer to being complete and we really start thinking about the public API though. There's some tricky questions about to what extent it's a C library vs a Python library when used in pure Cython mode.

Reply:
> in Python it feels more natural to import a package and be able to access subpackages/submodules without needing to do a separate import, wouldn't you say?

yes/no? scipy used to require you to explicitly import subpackages (although now they do on-demand lazy-loading so it's no longer the case). Some large packages I've worked on have had subpackage setups with an "import-what-you-need" approach for namespacing and to reduce import times.

But agreed we don't need to do anything now.